django/src/rdwatch/tasks/__init__.py

```diff
@@ -641,3 +643,48 @@ def generate_image_embedding(id: int):
     except Exception as e:
         # Handle exceptions (e.g., logging, showing error messages)
         print(f'Error processing image {site_image.id}: {e}')
+
+
+@signals.worker_ready.connect
+def download_sam_model_if_not_exists(**kwargs):
+    file_path = settings.SAM_CHECKPOINT_MODEL
+    logger.warning('Trying to download SAM')
+
+    # Check if the file exists
+    if not os.path.exists(file_path):
+        # If file doesn't exist, download it using requests
+        try:
+            url = 'https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth'
+            response = requests.get(url, stream=True)
+
+            # Check if the request was successful (status code 200)
+            if response.status_code == 200:
+                total_size = int(response.headers.get('content-length', 0))
+                bytes_downloaded = 0
+
+                with open(file_path, 'wb+') as file:
```

Comment:
```suggestion
                with open(file_path, 'wb') as file:
```

I'm not sure why I thought adding `+` here would solve anything - it turned out to be a different issue with my local system. Just using the `wb` mode should be fine, as the `+` just allows read access as well (which we don't need here and we're just writing the file). Sorry for the confusion :sweat_smile: 

(`open` docs for reference: https://docs.python.org/3/library/functions.html#open)